input/drivers/linuxraw_input.c

```diff
@@ -165,9 +166,58 @@ static void linuxraw_input_free(void *data)
       return;
 
    linux_terminal_restore_input();
+   linux_close_illuminance_sensor(linuxraw->illuminance_sensor);
    free(data);
 }
 
+static bool linuxraw_input_set_sensor_state(void *data, unsigned port, enum retro_sensor_action action, unsigned rate)
+{
+   linuxraw_input_t *linuxraw = (linuxraw_input_t*)data;
+
+   if (!linuxraw)
+      return false;
+
+   switch (action)
+   {
+      case RETRO_SENSOR_ILLUMINANCE_DISABLE:
+         /* If already disabled, then do nothing */
+         linux_close_illuminance_sensor(linuxraw->illuminance_sensor); /* noop if NULL */
+         linuxraw->illuminance_sensor = NULL;
+      case RETRO_SENSOR_GYROSCOPE_DISABLE:
+      case RETRO_SENSOR_ACCELEROMETER_DISABLE:
+         /** Unimplemented sensor actions that probably shouldn't fail */
+         return true;
+
+      case RETRO_SENSOR_ILLUMINANCE_ENABLE:
+         if (linuxraw->illuminance_sensor)
+            /* If the light sensor is already open, just set the rate */
+            linux_set_illuminance_sensor_rate(linuxraw->illuminance_sensor, rate);
+         else
+            linuxraw->illuminance_sensor = linux_open_illuminance_sensor(rate);
+
+         return linuxraw->illuminance_sensor != NULL;
+      default:
+         return false;
```

Comment:
This function should ALWAYS return something at the end. See previous comment about this.